### PR TITLE
Use multiprocessing instead of psutil to get cpu count

### PR DESCRIPTION
**Pull request by fonnesbeck**
`psutil` gives the wrong count on some architectures. This PR just uses `multiprocessing`.

### CODE REVIEW

**Review by eigenfoo — COMMENTED** [review]
Was there any reason we used `psutil` in the first place? It's just that we currently try `psutil`, and then fall back to `multiprocessing` if that doesn't work.

File: pymc3/sampling.py [hunk 1]
```diff
@@ -173,16 +173,10 @@ def _cpu_count():
     If not, we use the number provided by multiprocessing, but assume
```
Comment (twiecki) [inline pymc3/sampling.py]: comment above still mentiones psutils.

**Comment by twiecki** [comment]
Good question, the original PR https://github.com/pymc-devs/pymc3/pull/2613 by @aseyboldt does not explain why `psutil` was being used and there is no discussion on it. Could it be python 2 compatibility?

@fonnesbeck Needs a rebase.

**Comment by aseyboldt** [comment]
I used psutil to get the number of cores, multiprocessing only gives us the number of hardware threads. In general I guess it makes sense to only count cores (I did a bit of experimentation on by machine, but this probably depends on the model and possibly the hardware).

On January 18, 2019 10:11:44 AM CET, Thomas Wiecki <notifications@github.com> wrote:
>Good question, the original PR
>https://github.com/pymc-devs/pymc3/pull/2613 by @aseyboldt does not
>explain why `psutil` was being used and there is no discussion on it.
>Could it be python 2 compatibility?
>
>@fonnesbeck Needs a rebase.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pymc-devs/pymc3/pull/3344#issuecomment-455476996

-Adrian

**Comment by eigenfoo** [comment]
I guess it makes sense to only use `psutil` then? I also resolved the merge conflict.

**Comment by twiecki** [comment]
I think we should just leave it as is.

**Comment by ericmjl** [comment]
Would be good to in-line document why psutil is used, perhaps?

**Comment by fonnesbeck** [comment]
I have 2 dfifferent machines where `psutil` gets the count wrong, and I end up sampling with a single core.

**Comment by twiecki** [comment]
@fonnesbeck Wrong as in the count is 1?

**Comment by fonnesbeck** [comment]
Yes, using an intel i5, and `psutil.cpu_count(False)` returns 1.

**Comment by fonnesbeck** [comment]
Happy to keep using `psutil` but perhaps it should use `psutil.cpu_count(True) // 2`

**Comment by michaelosthege** [comment]
> Happy to keep using `psutil` but perhaps it should use `psutil.cpu_count(True) // 2`

If we want the number of cores, instead of the number of logical threads, this could be more robust:

```python
cpus = max(psutil.cpu_count(logical=False), psutil.cpu_count(logical=True) // 2)
```

**Comment by twiecki** [comment]
@fonnesbeck Still want to get this merged or should we close?

**Comment by fonnesbeck** [comment]
Let me try and figure it out today.

**Comment by fonnesbeck** [comment]
Didn't realize `psutil` is a third party package. Probably not worth having an extra dependency just for this.

**Comment by fonnesbeck** [comment]
Unrecoverable rebase/merge issues. Will close and create new PR now.

**Pull request closed without merging**